code/Items/Weapons/Knife.cs

```diff
@@ -70,19 +70,41 @@ private void MeleeAttack( float damage, float range, float radius )
 		if ( !IsServer )
 			return;
 
-		var damageInfo = DamageInfo.Generic( damage )
+		// Damage starts at 50 and multiplies to 100 on a backstab.
+		var damageInfo = DamageInfo.Generic( 50 )
 			.WithPosition( trace.EndPosition )
 			.UsingTraceResult( trace )
 			.WithAttacker( Owner )
 			.WithWeapon( this )
 			.WithFlag( DamageFlags.Slash );
 
-		if ( trace.Entity is Player player )
+		if ( trace.Entity is Player otherPlayer )
 		{
-			player.DistanceToAttacker = 0;
+			otherPlayer.DistanceToAttacker = 0;
 			PlaySound( FleshHit );
-			Owner.Inventory.DropActive();
-			Delete();
+
+			// Discard all z values to simplify to 2D.
+			var toTarget = (otherPlayer.Position - Owner.Position).WithZ( 0 ).Normal;
+			var ownerForward = Owner.EyeRotation.Forward.WithZ( 0 ).Normal;
+			var targetForward = otherPlayer.EyeRotation.Forward.WithZ( 0 ).Normal;
+
+			// Get all the dot products.
+			var behindDot = Vector3.Dot( toTarget, targetForward );
+			var facingDot = Vector3.Dot( toTarget, ownerForward );
+			var viewDot = Vector3.Dot( targetForward, ownerForward );
+
+			// Determine if it's a backstab.
```

Comment:
I'd remove this comment, it doesn't add anything.

Instead, maybe explain the reasoning behind the 0.0f, 0.5f, and 0.3f? Is there something you referenced?

Reply:
As potentially scandalous as it may be, I'm referencing the equation and values used in the TF2 spy knife code.
https://github.com/mastercomfig/team-comtress-2/blob/main/src/game/shared/tf/tf_weapon_knife.cpp#L432

I would prefer a way to simplify it so it can be convar-ified to make backstabs harder/easier. Maybe if I took the time to understand the vector math.